error/s2n_errno.h

```diff
@@ -297,6 +297,7 @@ typedef enum {
     S2N_ERR_EARLY_DATA_NOT_ALLOWED,
     S2N_ERR_NO_CERT_FOUND,
     S2N_ERR_CERT_NOT_VALIDATED,
+    S2N_ERR_CERT_REJECTED,
     S2N_ERR_NO_PRIVATE_KEY,
```

Comment:
Sorry, one more comment on your error :)

Should it be a usage error? I'm leaning more towards protocol error, because it it's triggered by peer input even if it does involve custom application logic.

Reply:
Oh okay yeah, that makes sense.